tests/test_hubspot_pagination_test.py

```diff
@@ -1,59 +1,102 @@
+from client import TestClient
 from base import HubspotBaseTest
 
 
 class TestHubspotPagination(HubspotBaseTest):
-    def name(self):
+
+    # TODO | Optimization is possible by checking only the first page in our test client
+    #        reads for 'has-more/hasMore'. That way if there are 1000 records and page size is 100,
+    #        we only get 100 of those records.
+    #        NOTE: This ^ approach makes assumptions about records returned are actually unique. May
+    #              not be true of events streams.
+
+    @staticmethod
+    def name():
         return "tap_tester_hubspot_pagination_test"
 
-    def expected_page_size(self):
+    def get_properties(self):
         return {
-            # "subscription_changes": 10 - 1000,
-            "subscription_changes": 1000,
-            # "email_events": 10 - 1000,
-            "email_events": 1000,
-            # "forms": ??, #infinity
-            # "workflows": ??,
-            # "owners": ??,
-            # "campaigns": ??,
-            # "campaigns": 500, # Can't make test data
-            # "contact_lists": 20 - 250,  #count,
-            # "deal_pipelines": ?? , # deprecated
-            # "contacts_by_company": 100,  #count # deprecated
-            "contact_lists": 250,
-            "contacts": 100, #count
-            "companies": 250,
-            "deals": 100,
-            "engagements": 250,
+            'start_date' : datetime.strftime(datetime.today()-timedelta(days=7), self.START_DATE_FORMAT)
         }
 
+    def setUp(self):
+        self.maxDiff = None  # see all output in failure
+        set_up_start = time.perf_counter()
+
+        self.my_timestamp = self.get_properties()['start_date']
+        test_client = TestClient(self.my_timestamp)
+        existing_records = dict()
+        streams = self.expected_streams() - {'email_events'} # we get this for free with subscription_changes
+        limits = self.expected_page_limits()
+         # 'contacts_by_company' stream needs to get companyIds first so putting the stream last in the list
+        stream_to_run_last = 'contacts_by_company'
+        if stream_to_run_last in streams:
+            streams.remove(stream_to_run_last)
+            streams = list(streams)
+            streams.append(stream_to_run_last)
+
+        for stream in streams:
+            # Get all records
+            if stream == 'contacts_by_company':
+                company_ids = [company['companyId'] for company in existing_records['companies']]
+                existing_records[stream] = test_client.read(stream, parent_ids=company_ids)
+            elif stream in {'companies', 'contact_lists', 'subscription_changes', 'engagements'}:
+                existing_records[stream] = test_client.read(stream, since=self.my_timestamp)
+            else:
+                existing_records[stream] = test_client.read(stream)
+                # existing_records['subscription_changes'] = test_client.get_subscription_changes()  # see BUG_TDL-14938
```

Comment:
I think you can drop this, the bug is marked in other tests and it looks like you were still able to test it.